### PR TITLE
PgBouncer - upgrade to 1.17.0

### PR DESCRIPTION
**Pull request by guidoiaquinti**
## Description
- upgrade PgBouncer from 1.12.0 to 1.17.0 (see changelog [here](https://www.pgbouncer.org/changelog.html#pgbouncer-112x)). We are doing that by upgrading the default image to an official bitnami one. From the changelog I don't see any major breaking change.   
- add more fine grain readiness/liveness checks + lifecycle config
- various other fixes to align this deployment and service more close to the others we already have 

Release notes are available at: https://github.com/PostHog/posthog.com/pull/3717

## Type of change
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## How has this been tested?
CI ✅ 

## Checklist
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: charts/posthog/templates/pgbouncer-deployment.yaml [hunk 1]
```diff
@@ -82,13 +78,26 @@ spec:
...
             port: 6543
-          initialDelaySeconds: 15
-          periodSeconds: 20
+          failureThreshold: {{ .Values.pgbouncer.livenessProbe.failureThreshold }}
+          initialDelaySeconds: {{ .Values.pgbouncer.livenessProbe.initialDelaySeconds }}
+          periodSeconds: {{ .Values.pgbouncer.livenessProbe.periodSeconds }}
+          successThreshold: {{ .Values.pgbouncer.livenessProbe.successThreshold }}
+          timeoutSeconds: {{ .Values.pgbouncer.livenessProbe.timeoutSeconds }}
+
+        lifecycle:
```
Comment (hazzadous) [inline charts/posthog/templates/pgbouncer-deployment.yaml:96]: 🥳

Comment (hazzadous) [reply charts/posthog/templates/pgbouncer-deployment.yaml:96]: What's the default behaviour?

Comment (guidoiaquinti) [reply charts/posthog/templates/pgbouncer-deployment.yaml:96]: https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/

File: charts/posthog/templates/pgbouncer-deployment.yaml [hunk 2]
```diff
@@ -44,51 +47,53 @@ spec:
...
-        - name: DB_HOST
+        - name: POSTGRESQL_HOST
           value: {{ include "posthog.postgresql.host" . }}
-        - name: DB_PORT
+        - name: POSTGRESQL_PORT
           value: {{ include "posthog.postgresql.port" . | quote }}
-        - name: DB_NAME
+        - name: POSTGRESQL_DATABASE
           value: {{ include "posthog.postgresql.database" . }}
-        - name: LISTEN_PORT
```
Comment (guidoiaquinti) [inline charts/posthog/templates/pgbouncer-deployment.yaml:73]: Moved to `values.yaml`

File: charts/posthog/templates/pgbouncer-deployment.yaml [hunk 3]
```diff
@@ -18,6 +18,9 @@ spec:
     metadata:
       annotations:
         checksum/secrets.yaml: {{ include (print $.Template.BasePath "/secrets.yaml") . | sha256sum }}
+        {{- if .Values.pgbouncer.podAnnotations }}
+{{ toYaml .Values.pgbouncer.podAnnotations | indent 8 }}
+      {{- end }}
```
Comment (hazzadous) [inline charts/posthog/templates/pgbouncer-deployment.yaml:23]:
```suggestion
        {{- end }}
```

File: charts/posthog/templates/pgbouncer-deployment.yaml [hunk 4]
```diff
@@ -44,51 +47,55 @@ spec:
...
-        - name: MAX_CLIENT_CONN
-          value: "1000"
-        - name: POOL_MODE
-          value: "transaction"
 {{- if .Values.pgbouncer.env }}
 {{ toYaml .Values.pgbouncer.env | indent 8 }}
 {{- end }}
         readinessProbe:
           tcpSocket:
             port: 6543
```
Comment (hazzadous) [inline charts/posthog/templates/pgbouncer-deployment.yaml:78]: We haven't changed the readiness probe, do you know why it wasn't working previously as expected?

File: charts/posthog/templates/pgbouncer-deployment.yaml [hunk 5]
```diff
@@ -44,51 +47,55 @@ spec:
...
+          failureThreshold: {{ .Values.pgbouncer.livenessProbe.failureThreshold }}
+          initialDelaySeconds: {{ .Values.pgbouncer.livenessProbe.initialDelaySeconds }}
+          periodSeconds: {{ .Values.pgbouncer.livenessProbe.periodSeconds }}
+          successThreshold: {{ .Values.pgbouncer.livenessProbe.successThreshold }}
+          timeoutSeconds: {{ .Values.pgbouncer.livenessProbe.timeoutSeconds }}
+
+        lifecycle:
+          preStop:
+            exec:
+              command: ["/bin/sh", "-c", "killall -INT pgbouncer && sleep 60"]
```
Comment (hazzadous) [inline charts/posthog/templates/pgbouncer-deployment.yaml:97]: Do you mind adding a comment on what pgbouncer does on sig INT?

File: charts/posthog/templates/pgbouncer-deployment.yaml (hunk 4 above)
Comment (guidoiaquinti) [reply charts/posthog/templates/pgbouncer-deployment.yaml:78]: `initialDelaySeconds` is now 10 from the initial 5
`periodSeconds` is now 5 from the initial 10

I don't think we had problems before but this is a less strict config.

File: ci/kubetest/helpers/clickhouse.py [hunk 6]
```diff
@@ -3,7 +3,7 @@
 
 def get_clickhouse_statefulset_spec(kube):
     start = time.time()
-    while time.time() - start < 120:
+    while time.time() - start < 300:
```
Comment (fuziontech) [inline ci/kubetest/helpers/clickhouse.py:6]:
```suggestion
    while (time.time() - start) < 300:
```

File: ci/kubetest/helpers/clickhouse.py [hunk 7]
```diff
@@ -19,7 +19,7 @@ def get_clickhouse_statefulset_spec(kube):
 
 def get_clickhouse_cluster_service_spec(kube):
     start = time.time()
-    while time.time() - start < 120:
+    while time.time() - start < 300:
```
Comment (fuziontech) [inline ci/kubetest/helpers/clickhouse.py:22]:
```suggestion
    while (time.time() - start) < 300:
```
I'm just neurotic about clarity on order of ops.

File: charts/posthog/templates/pgbouncer-deployment.yaml (hunk 1 above)
Comment (fuziontech) [reply charts/posthog/templates/pgbouncer-deployment.yaml:100]: me gusta 🥳

**Review by fuziontech — APPROVED** [review]
Dude great work as usual.

🚢 it

**Pull request merged**